src/subdomains/supporting/support-issue/services/limit-request.service.ts

```diff
@@ -25,31 +25,36 @@ export class LimitRequestService {
     private readonly notificationService: NotificationService,
   ) {}
 
-  async increaseLimit(dto: LimitRequestDto, kycHash: string, userDataId?: number): Promise<void> {
+  async increaseLimit(dto: LimitRequestDto, kycHash?: string, userDataId?: number): Promise<void> {
     // get user data
     const user = userDataId
       ? await this.userDataService.getUserData(userDataId)
       : await this.userDataService.getByKycHashOrThrow(kycHash);
 
-    if (user.kycLevel < KycLevel.LEVEL_50) throw new BadRequestException('Missing KYC');
-
-    // create entity
-    let entity = this.limitRequestRepo.create(dto);
-    entity.userData = user;
-
     // upload document proof
     if (dto.documentProof) {
       const { contentType, buffer } = Util.fromBase64(dto.documentProof);
 
-      entity.documentProofUrl = await this.storageService.uploadFile(
+      const documentProofUrl = await this.storageService.uploadFile(
         user.id,
         FileType.USER_NOTES,
         `${Util.isoDateTime(new Date())}_limit-request_user-upload_${dto.documentProofName}`,
         buffer,
         contentType as ContentType,
       );
+
+      return void this.increaseLimitInternal({ ...dto, documentProofUrl }, user);
     }
 
+    return void this.increaseLimitInternal(dto, user);
+  }
```

Comment:
I think this doesn't wait 🤔  just use `await this.increaseLimitInternal(dto, user);`